app/uk/gov/hmrc/economiccrimelevyaccount/controllers/AccountController.scala

```diff
@@ -32,15 +34,30 @@ class AccountController @Inject() (
   val controllerComponents: MessagesControllerComponents,
   authorise: AuthorisedAction,
   enrolmentStoreProxyService: EnrolmentStoreProxyService,
-  view: AccountView
+  view: AccountView,
+  obligationDataConnector: ObligationDataConnector
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport {
 
   def onPageLoad: Action[AnyContent] = authorise.async { implicit request =>
-    enrolmentStoreProxyService.getEclRegistrationDate(request.eclRegistrationReference).map { registrationDate =>
-      Ok(view(request.eclRegistrationReference, ViewUtils.formatLocalDate(registrationDate)))
+    enrolmentStoreProxyService.getEclRegistrationDate(request.eclRegistrationReference).flatMap { registrationDate =>
+      obligationDataConnector
+        .getObligationData()
+        .map { obligationData =>
+          Ok(
+            view(
+              request.eclRegistrationReference,
+              ViewUtils.formatLocalDate(registrationDate),
+              getLatestObligation(obligationData)
+            )
+          )
+        }
     }
   }
 
+  private def getLatestObligation(obligationData: Option[ObligationData]): Option[ObligationDetails] = {
+   var test = obligationData.map(_.obligations.flatMap(_.obligationDetails.sortBy(_.inboundCorrespondenceDueDate))).flatMap(_.headOption)
```

Comment:
We should not use `var` in our code as it introduces mutable state. In 99.9% of cases using var in scala is bad practice and should be avoided unless absolutely necessary.

in this instance you probably don't even need to be assigning to a variable anyway, but if you do, always use `val` 🙂

Reply:
Done.